pygfx/renderers/wgpu/linerender.py

```diff
@@ -25,74 +54,382 @@ def compute_shader(
-    if positions2_nitems != positions1.nitems * 2:
-        # Create buffer that only exist on the GPU, we provide stub data so that
-        # the buffer knows type and strides (needed when used as vertex buffer)
-        geometry._wgpu_line_renderer_positions2 = Buffer(
-            nbytes=positions1.nitems * 2 * 4 * 4,
-            nitems=positions1.nitems * 2,
-            format="float4",
-            usage="vertex|storage",
-        )
-    positions2 = geometry._wgpu_line_renderer_positions2
+    primitive = wgpu.PrimitiveTopology.line_strip
+    if isinstance(material, LineThinSegmentMaterial):
+        primitive = wgpu.PrimitiveTopology.line_list
 
     return [
         {
-            "compute_shader": compute_shader,
-            "indices": (positions1.nitems, 1, 1),
+            "vertex_shader": vertex_shader_thin,
+            "fragment_shader": fragment_shader_thin,
+            "primitive_topology": primitive,
+            "indices": (positions1.nitems, 1),
+            "vertex_buffers": [positions1],
             "bindings0": {
-                0: (wgpu.BindingType.storage_buffer, positions1),
-                1: (wgpu.BindingType.storage_buffer, positions2),
-                2: (wgpu.BindingType.uniform_buffer, material.uniform_buffer),
+                0: (wgpu.BindingType.uniform_buffer, render_info.stdinfo),
+                1: (wgpu.BindingType.uniform_buffer, material.uniform_buffer),
             },
+            # "bindings1": {},
+            "target": None,  # default
         },
+    ]
+
+
+@register_wgpu_render_function(Line, LineMaterial)
+def line_renderer(wobject, render_info):
+    """ Render function capable of rendering lines.
+    """
+
+    material = wobject.material
+    geometry = wobject.geometry
+
+    assert isinstance(material, LineMaterial)
+
+    positions1 = geometry.positions
+
+    if isinstance(material, LineArrowMaterial):
+        vert_shader = vertex_shader_arrow
+        n = (positions1.nitems // 2) * 2 * 4
+    elif isinstance(material, LineSegmentMaterial):
+        vert_shader = vertex_shader_segment
+        n = (positions1.nitems // 2) * 2 * 5
+    else:
+        vert_shader = vertex_shader
+        n = positions1.nitems * 5
+
+    return [
+        # {
+        #     "compute_shader": compute_shader,
+        #     "indices": (positions1.nitems, 1, 1),
+        #     "bindings0": {
+        #         0: (wgpu.BindingType.storage_buffer, positions1),
+        #         1: (wgpu.BindingType.storage_buffer, positions2),
+        #         2: (wgpu.BindingType.uniform_buffer, material.uniform_buffer),
+        #     },
+        # },
         {
-            "vertex_shader": vertex_shader,
+            "vertex_shader": vert_shader,
             "fragment_shader": fragment_shader,
             "primitive_topology": wgpu.PrimitiveTopology.triangle_strip,
-            "indices": positions1.nitems * 2,
-            "vertex_buffers": [positions2],
+            "indices": (n, 1),
```

Comment:
So you are just setting up a render pass with `n` iterations without an actual vertex buffer being assigned? Clever!! Was this possible in opengl too?

Reply:
Yes, since OpenGL 4.6 apparently. But one could always simulate it with textures and a single vertexbuffer containing indices.

Reply:
Right, i see. Anyway, pretty cool! I can see this being used to implement mesh slicing too, perhaps. Is it possible to discard vertices? Probably not eh?